Library/brew.sh

```diff
@@ -57,7 +57,7 @@ then
   unset HOMEBREW_RUBY_PATH
 fi
 
-HOMEBREW_SYSTEM="$(uname -s)"
+HOMEBREW_SYSTEM="$(/usr/bin/uname -s)"
```

Comment:
Yes, I have neglected the Linux side of things and that we might want to run a vanilla `brew` under Linux soon (e.g. for testing). Not good.

Introducing shell functions to wrap a bunch of binaries feels somewhat dirty to me. Maybe backing up the user's `PATH` in a variable and then overriding it with a known-sane value (per platform) is a better idea to avoid unexpected inference from the user's `PATH`?